### ISSUE TITLE
[Question] Plotting null values in XYFrame

### ISSUE THREAD

**Issue by bengarvey** [emeeks/semiotic#35]
![screen shot 2017-09-16 at 11 25 40 am](https://user-images.githubusercontent.com/1342331/30513470-d167e5ea-9ad1-11e7-9573-4ecb900b2b5d.png)

For the green and red lines in this chart, I don't have data prior to 1982, so those values are null in my dataset. Semiotic still plots them at the bottom of the y axis. I tried making the line formatting conditional to hide them, but that didn't seem to work.

Is this considered a bug or is there a way around it?

**Comment by emeeks** [comment]
You can use the `defined` attribute to set rules for when a value means a gap. So in your case you should be able to set:
`defined={d => d.yourValue !== null}`
If that doesn't work let me know. That will work for any gaps in the data.
Love the sketchy rendering in looks so hand-drawn.

**Comment by bengarvey** [comment]
💥 
![screen shot 2017-09-16 at 4 02 11 pm](https://user-images.githubusercontent.com/1342331/30515571-7926d9e6-9af8-11e7-866e-c02c906fc8e3.png)

That worked perfecly, thanks.

**Comment by bengarvey** [comment]
@emeeks Upgraded to 1.16.3 and I think this is broken again. 

With this code:
`defined={d => d.yourValue !== null}`

It seems to ignore the nulls completely instead of skipping them, making it span the data across the whole chart even where it shouldn't be. I fixed this by explicitly setting the xExtent, which solved the problem, but I figured I'd let you know this behavior changed. 

The problem explained visually:
With no xExtent set, but null values for earlier datapoints.
![screen shot 2019-01-06 at 10 08 59 pm](https://user-images.githubusercontent.com/1342331/50747038-d3d82000-11ff-11e9-9c38-d3a373590a09.png)

With explicit xExtent set, correctly not plotting the nulls.
![screen shot 2019-01-06 at 10 09 10 pm](https://user-images.githubusercontent.com/1342331/50747057-eb170d80-11ff-11e9-9e71-40675d358659.png)
